### PR TITLE
CLN - clean repo

### LINKED ISSUE #1: CLN readme

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
- [x] clean README
- [x] pass CI tests

--------------

closes #1

### CODE REVIEW

**Comment by Badr-MOUFAD** [comment]
CI fails because when importing `benchmark_utils`, numba is still not installed yet. https://github.com/benchopt/benchmark_sqrt_lasso/actions/runs/3650124015/jobs/6165709504 

@mathurinm any ideas on what might be the cause?

File: solvers/pdcd_ws.py [hunk 1]
```diff
@@ -12,8 +12,9 @@
 class Solver(BaseSolver):
     name = "PDCD-WS"
 
+    install_cmd = 'conda'
     requirements = [
-        'pip:git+https://github.com/Badr-MOUFAD/skglm.git@pdcd-ws-skglm'
+        'pip:git+https://github.com/Badr-MOUFAD/skglm.git@pmain'
```
Comment (mathurinm) [inline solvers/pdcd_ws.py]:
```suggestion
        'pip:git+https://github.com/Badr-MOUFAD/skglm.git@main'
```

**Comment by mathurinm** [comment]
Can you use a safe_import_context ?

File: solvers/pdcd_ws.py (hunk 1 above)
Comment (Badr-MOUFAD) [reply solvers/pdcd_ws.py]: nice catch indeed!

**Comment by Badr-MOUFAD** [comment]
> Can you use a safe_import_context ?

Tried it but it doesn't solve the problem.

-----------
P.S: importing under `safe_import_context` seem to not work outside the solvers, datasets, and objective https://github.com/benchopt/benchopt/issues/509

**Comment by Badr-MOUFAD** [comment]
CI test fails for the prox newton solver, in particular for [the convergence test](https://github.com/benchopt/benchmark_sqrt_lasso/actions/runs/3658103660/jobs/6182625583#step:6:514), since we interrupt the solver when [the residuals get small](https://github.com/scikit-learn-contrib/skglm/blob/4bee85f9ec1a807307b96bcf595339fd9874460f/skglm/experimental/sqrt_lasso.py#L35-L49).

WDYT @mathurinm?

**Comment by Badr-MOUFAD** [comment]
There is still an issue with imports in ``benchmark_utils`` folder.
Better we move it to an issue and tackle it through a new PR.

Merging as the PR brings substantial enhancement to the repo

**Pull request merged**